File: InPlan.Web/Modules/ServerTypes/Parametros/ClientesRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface ClientesRow {
    IdCliente?: number;
    IdEmpresa?: number;
    IdArea?: number;
    DescripcionCorta?: string;
    Descripcion?: string;
    Observaciones?: string;
    Baja?: boolean;
    IdEmpresaDescripcionCorta?: string;
    IdAreaDescripcion?: string;
}

export abstract class ClientesRow {
    static readonly idProperty = 'IdCliente';
    static readonly nameProperty = 'DescripcionCorta';
    static readonly localTextPrefix = 'Parametros.Clientes';
    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<ClientesRow>();
}

File: InPlan.Web/Modules/Parametros/Clientes/ClientesRow.cs
using InPlan.Tenencia;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace InPlan.Parametros;

[ConnectionKey("Default"), Module("Parametros"), TableName("Clientes")]
[DisplayName("Cliente"), InstanceName("Cliente")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
public sealed partial class ClientesRow : Row<ClientesRow.RowFields>, IIdRow, INameRow
{
    const string jIdEmpresa = nameof(jIdEmpresa);
    const string jIdArea = nameof(jIdArea);

    class RowTemplate
    {
        [DisplayName("Id Cliente"), Identity, IdProperty]
        public int? IdCliente { get; set; }

        [LookupEditor(typeof(EmpresasRow))]
        [DisplayName("Empresa"), NotNull, ForeignKey("Empresas", "IdEmpresa"), LeftJoin(jIdEmpresa), TextualField(nameof(IdEmpresaDescripcionCorta))]
        public int? IdEmpresa { get; set; }

        [LookupEditor(typeof(AreasRow), CascadeFrom = "IdEmpresa")]
        [DisplayName("Área"), NotNull, ForeignKey("Areas", "IdArea"), LeftJoin(jIdArea), TextualField(nameof(IdAreaDescripcion))]
        public int? IdArea { get; set; }

        [DisplayName("Descripción Corta"), Size(200), NotNull, QuickSearch, NameProperty]
        public string DescripcionCorta { get; set; }

        [DisplayName("Descripción")]
        public string Descripcion { get; set; }

        [DisplayName("Observaciones")]
        public string Observaciones { get; set; }

        [DisplayName("Baja"), NotNull]
        public bool? Baja { get; set; }

        [DisplayName("Empresa"), Expression($"{jIdEmpresa}.[DescripcionCorta]")]
        public string IdEmpresaDescripcionCorta { get; set; }

        [DisplayName("Área"), Expression($"{jIdArea}.[Descripcion]")]
        public string IdAreaDescripcion { get; set; }
    }
}